packages/ses/test/test-anticipate-async-iterator-helpers-shimmed.js

```diff
@@ -0,0 +1,38 @@
+// KLUDGE HAZARD The core-js shims are written as sloppy code
+// and so introduce sloppy functions.
+import 'core-js/actual/async-iterator/index.js';
+import test from 'ava';
+import '../index.js';
+
+// KLUDGE HAZARD only for testing with the sloppy modules of the
+// core-js iterator shim.
+// We mutate the permits to tolerates the sloppy functions for testing
+// by sacrificing security. The caller and arguments properties of
+// sloppy functions violate ocap encapsulation rules.
+import { AsyncFunctionInstance } from '../src/permits.js';
+
+AsyncFunctionInstance.arguments = {};
+AsyncFunctionInstance.caller = {};
```

Comment:
Wait, why do we have a sloppy `AsyncFunctionInstance`?

Reply:
(Deleted my comment as confused. I'll investigate)

Reply:
Trying this out on `test-anticipate-iterator-helpers-shimmed.js`. Commenting out the monkey patching of the corresponding two items from the permits list, 
```js
// FunctionInstance.arguments = {};
// FunctionInstance.caller = {};
```
and rerunning the test, I get
```
Removing intrinsics.Iterator.from.arguments
  ✘ [fail]: shimmed iterator helpers Error thrown in test
failed to delete intrinsics.Iterator.from.arguments (TypeError#1)
TypeError#1: Cannot delete property 'arguments' of function from(O) {
    var iteratorRecord = getIteratorFlattenable(typeof O == 'string' ? toObject(O) : O);
   ...<omitted>... }
    var iteratorRecord = getIteratorFlattenable(typeof O == 'string' ? toObject(O) : O);
   ...<omitted>... }
  at visitProperties (file:///Users/markmiller/src/ongithub/endojs/endo/packages/ses/src/permits-intrinsics.js:289:22)
```

What happens is that core-js adds `Iterator.from` as you'd expect. But the function it adds is a sloppy function that, on v8/node, has *undeletable* `arguments` and `caller` properties, which is not surprising. Our permit processing correctly fails safe when it finds it cannot delete them. Fortunately, the value of these properties at the time of permit processing is null.

By adding these two properties to the permits record that applies to most functions, we allow most functions in general to have these extra properties, including the ones added by the core-js shim.

Reply:
Resolving